Matter_Engine/Query/Density.cs

```diff
@@ -51,6 +51,12 @@ public static partial class Query
         [Output("density", "The density of the material. Additional info on how the value has been acquired is recorded in the warning", typeof(Density))]
         public static double Density(this Material material, Type type = null, double tolerance = 0.001)
         {
+            if(material == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot query the density of a null material.");
+                return 0;
```

Comment:
shouldn't it be `double.NaN`? This is a general question regarding all methods returning `double`

Reply:
`double.NaN` would also need to be picked up as a check by anything using the method (which LCA does quite a bit @michaelhoehn ). I opted for 0 here just because if you have no material, you have no density rather than an invalid density.

I'm happy either way, but would not like to spend 3 weeks debating it so if we're doing a change let's go for a change (with discussion if needed) but if not then we can go with what's here :wink: